pinot-common/src/main/java/com/linkedin/pinot/common/segment/fetcher/GenericSegmentFetcher.java

```diff
@@ -0,0 +1,58 @@
+/**
+ * Copyright (C) 2014-2018 LinkedIn Corp. (pinot-core@linkedin.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.linkedin.pinot.common.segment.fetcher;
+
+import com.linkedin.pinot.filesystem.PinotFS;
+import com.linkedin.pinot.filesystem.PinotFSFactory;
+import java.io.File;
+import java.net.URI;
+import java.util.Collections;
+import java.util.Set;
+import org.apache.commons.configuration.Configuration;
+
+
+public class GenericSegmentFetcher implements SegmentFetcher {
+  private PinotFS _pinotFS;
+
+  @Override
+  public void init(Configuration config) {
+  }
+
+  public GenericSegmentFetcher() {
+
+  }
+
+  public GenericSegmentFetcher(URI uri, Configuration configuration) throws Exception {
+    PinotFSFactory factory = new PinotFSFactory(configuration);
+    _pinotFS = factory.create(uri);
```

Comment:
Should this class create and hold the FS upfront? Or should it be flexible enough that it can use the appropriate FS per invocation? If latter, we would need to avoid creating FS object per invocation of fetch.

Reply:
Discussed this offline, this isn't possible at this point if we want backwards compatibility